fjord/suggest/providers/trigger/models.py

```diff
@@ -0,0 +1,253 @@
+import re
+
+from django.db import models
+from django.utils.lru_cache import lru_cache
+
+from product_details import product_details
+
+from fjord.base.models import ModelBase
+from fjord.feedback.models import Product
+
+
+def _lang_native(locale):
+    """Look up locale code for native word
+
+    Note: This is case-sensitive. e.g. "en-US" matches English, but
+    "en-us" matches nothing.
+
+    :arg locale: locale string
+
+    :returns: language native name for that locale
+    """
+    try:
+        return product_details.languages[locale]['native']
+    except KeyError:
+        return locale
+
+
+class TriggerRuleManager(models.Manager):
+    def all_enabled(self):
+        return self.filter(is_enabled=True).order_by('sortorder')
+
+
+@lru_cache()
+def _generate_keywords_regex(keywords):
+    """Generates a Regex object to compare with
+
+    * splits keywords into parts
+    * for each part
+      * converts to unicode
+      * converts white space to '\s+'
+    * joins the parts to create a middle
+    * joins that with empty string bounds
+    * compiles it into a Regex and returns it
+
+    :arg keywords: list of words and phrases to match
+
+    :returns: regex object
+    """
+    assert isinstance(keywords, unicode)
+
+    def _escape(utext):
```

Comment:
Actually, `re.escape` doesn't do what I want. It escapes everything that's not an ascii alphanumeric character which makes a mess of things because I want to do different things with spaces and it doesn't work well with utf-8.

This probably isn't perfect, but it covers all the characters I suspect someone writing a trigger rule would need covered.